### PR TITLE
Fix bug with direct upload

### LINKED ISSUE #1146: "Agree terms" validation must be done on back end side
Rtmedia > Other settings > Ask user to agree terms 

image

Step to generate this scenario ->
Now delete the element `<div id="rtm-file_upload-ui" class="rtm-tab-content">` using any debugging tool
and then upload the file. 
image

It would allow user to upload file without checking the agree option.

### PR DESCRIPTION
**Pull request by SID177**
- Fix bug with direct upload
Fixes #1146

### CODE REVIEW

File: app/assets/js/rtm-upload-terms.js [hunk 1]
```diff
@@ -76,28 +76,53 @@ if ( 'object' === typeof rtMediaHook ) {
...
             }
         }
 
+        if ( ! isTermsEnabled ) {
+            return args;
+        }
+
+        if ( terms.length === 0 ) {
+            rtp_display_terms_warning( parent, rtmedia_upload_terms_data.message );
+
```
Comment (abhijitrakas) [inline app/assets/js/rtm-upload-terms.js]:
```suggestion
```

File: app/assets/js/rtm-upload-terms.js [hunk 2]
```diff
@@ -76,28 +76,53 @@ if ( 'object' === typeof rtMediaHook ) {
...
+            if ( 'true' === rtmedia_upload_terms_data.activity_terms_enabled ) {
+                isTermsEnabled = true;
             }
         }
 
+        if ( ! isTermsEnabled ) {
+            return args;
+        }
+
+        if ( terms.length === 0 ) {
```
Comment (abhijitrakas) [inline app/assets/js/rtm-upload-terms.js]: Cloud we need to check for `terms` variable?

Comment (abhijitrakas) [reply app/assets/js/rtm-upload-terms.js]: Yoda

File: app/assets/js/rtm-upload-terms.js [hunk 3]
```diff
@@ -76,28 +76,53 @@ if ( 'object' === typeof rtMediaHook ) {
...
+
+            return false;
+        }
+
+        rtmediaTermsConditionsElement = terms;
+        if ( terms.prop( 'checked' ) ) {
+            return true;
+        } else {
+            rtp_display_terms_warning( parent, rtmedia_upload_terms_data.message );
+
```
Comment (abhijitrakas) [inline app/assets/js/rtm-upload-terms.js]:
```suggestion
```

File: app/assets/js/rtm-upload-terms.js [hunk 4]
```diff
@@ -76,28 +76,53 @@ if ( 'object' === typeof rtMediaHook ) {
...
-                return false;
+            if ( 'true' === rtmedia_upload_terms_data.uploader_terms_enabled ) {
+                isTermsEnabled = true;
             }
+        } else if ( 'activity' === args.src ) {
 
-            rtmediaTermsConditionsElement = terms;
-            if ( terms.prop( 'checked' ) ) {
-                return true;
+            if ( terms.length === 0 ) {
```
Comment (abhijitrakas) [inline app/assets/js/rtm-upload-terms.js]: Yoda and `terms` variable check?

File: app/assets/js/rtm-upload-terms.js (hunk 2 above)
Comment (SID177) [reply app/assets/js/rtm-upload-terms.js]: We're using jquery so no need to check `terms` variable, jQuery won't return `undefined`.

File: app/assets/js/rtm-upload-terms.js (hunk 1 above)
Comment (SID177) [reply app/assets/js/rtm-upload-terms.js]: Don't we leave blank line before `return`?

Comment (abhijitrakas) [reply app/assets/js/rtm-upload-terms.js]: Need but not in just two-line `if` block condition.

Comment (SID177) [reply app/assets/js/rtm-upload-terms.js]: Okay

File: app/assets/js/rtm-upload-terms.js (hunk 2 above)
Comment (SID177) [reply app/assets/js/rtm-upload-terms.js]: Sorry, `length` variable does change on assigning. I got it mixed with `Java`

**Pull request merged**